lib/src/screenshots.dart

```diff
@@ -0,0 +1,327 @@
+  if (convertWithCwebpProblems.isEmpty) {
+    return [];
+  }
+  problems.addAll(convertWithCwebpProblems);
+
+  final gif2webpProblems = await _convertGifToWebp(originalPath, webpPath);
+  if (gif2webpProblems.isEmpty) {
+    return [];
+  }
+  problems.addAll(gif2webpProblems);
+  problems.add(
+      'Generating webp image for $originalPath failed due to invalid input');
+  return problems;
+}
+
+Future<List<String>> _copyIfAlreadyWebp(
+    String originalPath, String webpPath) async {
+  final infoResult = await _checkedRunProc(['webpinfo', originalPath]);
+  if (infoResult.exitCode == 0) {
+    await File(originalPath).copy(webpPath);
+    return [];
+  }
+  return [
+    '$originalPath: Tried interpreting screenshot as WebP. Failed with exitcode ${infoResult.exitCode}.'
+  ];
+}
+
+Future<List<String>> _convertWithCWebp(
+    String originalPath, String webpPath) async {
+  final cwebpResult =
+      await _checkedRunProc(['cwebp', originalPath, '-o', webpPath]);
+  if (cwebpResult.exitCode == 0) {
+    return [];
+  }
+  return [
+    '$originalPath: Converting screenshot with `cwebp` failed with exitcode ${cwebpResult.exitCode}.'
+  ];
+}
+
+Future<List<String>> _convertGifToWebp(
+    String originalPath, String webpPath) async {
+  final gif2webpResult =
+      await _checkedRunProc(['gif2webp', originalPath, '-o', webpPath]);
+
+  if (gif2webpResult.exitCode == 0) {
+    return [];
+  }
+  return [
+    '$originalPath: Tried interpreting screenshot as GIF. Failed with exitcode ${gif2webpResult.exitCode}.'
+  ];
+}
+
+/// Generates a PNG and a WebP thumbnail from the WebP screenshot found at
+/// [webpPath] and writes them to [pngThumbnailPath] and [webpThumbnailPath]
+/// respectively.
+///
+/// Returns a list of problems if generating the thumbnails did not succeed.
+/// On success, an empty list is returned.
+//
+/// If the WebP screenshot is animated, the `webpmux` tool is used to extract
+/// the first frame for the thumbnail.
+///
+/// The `webpinfo` tool is used to determine the original height and width. The
+/// `cwebp` converter is used for resizing the screenshot. The `dwebp` converter
+/// is used for generating the thumbnail in PNG.
+///
+/// Thumbnail generation will fail if any of the mentioned tools are not
+/// available on the command line and an error message will be written to stderr.
+Future<List<String>> _generateThumbnails(String originalPath, String webpPath,
+    String webpThumbnailPath, String pngThumbnailPath) async {
+  late String staticWebpPath;
+  final infoResult = await runProc(['webpinfo', webpPath]);
+  if (infoResult.exitCode != 0) {
+    return [
+      "$originalPath: Running 'webpinfo' on $webpPath failed with exitcode ${infoResult.exitCode}."
+    ];
+  }
+
+  if ((infoResult.stdout as String).contains('Animation: 1')) {
+    staticWebpPath = path.join(Directory.systemTemp.path,
+        '${path.basenameWithoutExtension(webpPath)}_static.webp');
```

Comment:
Another temporary folder that is leaked.

Reply:
Done.